Solutions/Corvus.Testing.Tenancy/Corvus/Testing/Tenancy/TenantLease.cs

```diff
@@ -0,0 +1,47 @@
+﻿// <copyright file="TenantLease.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Corvus.Testing.Tenancy
+{
+    using System;
+    using System.Threading;
+    using Corvus.Leasing;
+    using Corvus.Tenancy;
+
+    /// <summary>
+    /// A tenant and the associated lease.
+    /// </summary>
+    public class TenantLease
+    {
+        private readonly ITenant tenant;
+
+        /// <summary>
+        /// A tenant and its associated lease.
+        /// </summary>
+        /// <param name="tenant">The tenant that has been leased.</param>
+        /// <param name="lease">The lease for the tenant, that must be released when the tenant is no longer in use.</param>
+        /// <param name="cts">The cancellation token source for releasing the lease.</param>
+        internal TenantLease(ITenant tenant, Lease lease, CancellationTokenSource cts)
+        {
+            this.Lease = lease ?? throw new ArgumentNullException(nameof(lease));
+            this.tenant = tenant ?? throw new ArgumentNullException(nameof(tenant));
+            this.CancellationTokenSource = cts;
+        }
+
+        /// <summary>
+        /// Gets the leased tenant.
+        /// </summary>
+        public ITenant Tenant { get; }
+
+        /// <summary>
+        /// Gets the lease for the tenant which must be released when the tenant is no longer required.
+        /// </summary>
+        public Lease Lease { get; }
+
+        /// <summary>
+        /// Gets the cancellation token source to use to release the renewing lease.
+        /// </summary>
+        internal CancellationTokenSource CancellationTokenSource { get; }
```

Comment:
Consider adding some documentation around ownership of this. This is a disposable resource, and it does get disposed, but it's not immediately obvious how. It relies on `ReleaseLeaseAsync` getting called at some point, which happens when this is passed to `ReleaseWellKnownTestTenant`. (So there's nothing wrong, it's just that if anyone else, e.g., future me, looks at this and wonders whether disposal of this is correctly handled, I'd like to save them/me the few minutes of thinking this through that I just went through.)